### PR TITLE
added redis config to leaderboard and refactored leaderboard module

### PR DESCRIPTION
**Pull request by lucas-machado**
Now clients of the podium library can instantiate a client without managing Redis themselves. Refactoring for this also brought the opportunity to make the API cleaner: we no longer need to create a leaderboard every time an operation is needed. We use a single leaderboard client that gives us all leaderboard functionality and redis management.

### CODE REVIEW

File: api/healthcheck_test.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ var _ = Describe("Healthcheck Handler", func() {
 
 	It("Should fail if redis failing", func() {
 		a := GetDefaultTestApp()
-		a.RedisClient.Client = GetFaultyRedis()
+		a.Leaderboards.RedisClient().Client = GetFaultyRedis()
```
Comment (cscatolini) [inline api/healthcheck_test.go]: Should we expose the inner client and allow it to be changed like this?

Comment (lucas-machado) [reply api/healthcheck_test.go]: Makes sense that we don't. I'll change the test code to allow for the removal of this method.

Comment (cscatolini) [reply api/healthcheck_test.go]: Great! Thanks

**Comment by cscatolini** [comment]
Great job!

**Pull request merged**